src/FilenameParsing/FileIDHelper.php

```diff
@@ -53,4 +53,11 @@ public function isVariantOf($fileID, ParsedFileID $parsedFileID);
      * @return string
      */
     public function lookForVariantIn(ParsedFileID $parsedFileID);
+
+    /**
+     * Specify if this File ID Helper stores variants in subfolders and require a recursive look up to find all
+     * variants.
+     * @return bool
+     */
+    public function lookForVariantRecursive(): bool;
```

Comment:
What's the existing unit test coverage for this like?

Reply:
`lookForVariantIn` and the logic for finding variants has pretty solid coverage. 

`lookForVariantRecursive` is a new method so it won't have any coverage. I can add some unit test of it, but the expectation is that each file ID helper will always return the same value for this method.

Reply:
Sweet, I'm going to presume `lookForVariantIn` means that all the variants are in sane places e.g. hash and natural only have variants in the same folder as the main file

It's almost pointless adding unit tests to `lookForVariantRecursive` except to make the the codecov level nicer.  Probably should just for the sake of completionism, though I'm honestly not fussed